pkg/interfaces/contracts/pool-gyro/IGyroECLPPool.sol

```diff
@@ -2,8 +2,75 @@
 
 pragma solidity ^0.8.24;
 
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
 import { IBasePool } from "../vault/IBasePool.sol";
 
+/**
+ * @notice Gyro E-CLP Pool data that cannot change after deployment.
+ * @param tokens Pool tokens, sorted in token registration order
+ * @param decimalScalingFactors Conversion factor used to adjust for token decimals for uniform precision in
+ * calculations. FP(1) for 18-decimal tokens
+ * @param paramsAlpha Lower price limit. alpha > 0
+ * @param paramsBeta Upper price limit. beta > alpha > 0
+ * @param paramsC `c = cos(-phi) >= 0`, rounded to 18 decimals. Phi is the rotation angle of the ellipse
+ * @param paramsS `s = sin(-phi) >= 0`, rounded to 18 decimals. Phi is the rotation angle of the ellipse
+ * @param paramsLambda Stretching factor, lambda >= 1. When lambda == 1, we have a perfect circle
+ * @param tauAlphaX
+ * @param tauAlphaY
+ * @param tauBetaX
+ * @param tauBetaY
+ * @param u from (A chi)_y = lambda * u + v
+ * @param v from (A chi)_y = lambda * u + v
+ * @param w from (A chi)_x = w / lambda + z
+ * @param z from (A chi)_x = w / lambda + z
+ * @param dSq error in c^2 + s^2 = dSq, used to correct errors in c, s, tau, u,v,w,z calculations
+ */
+struct GyroECLPPoolImmutableData {
+    IERC20[] tokens;
+    uint256[] decimalScalingFactors;
+    int256 paramsAlpha;
+    int256 paramsBeta;
+    int256 paramsC;
+    int256 paramsS;
+    int256 paramsLambda;
+    int256 tauAlphaX;
+    int256 tauAlphaY;
+    int256 tauBetaX;
+    int256 tauBetaY;
+    int256 u;
+    int256 v;
+    int256 w;
+    int256 z;
+    int256 dSq;
+}
+
+/**
+ * @notice Snapshot of current Gyro E-CLP Pool data that can change.
+ * @dev Note that live balances will not necessarily be accurate if the pool is in Recovery Mode. Withdrawals
+ * in Recovery Mode do not make external calls (including those necessary for updating live balances), so if
+ * there are withdrawals, raw and live balances will be out of sync until Recovery Mode is disabled.
+ *
+ * @param balancesLiveScaled18 Token balances after paying yield fees, applying decimal scaling and rates
+ * @param tokenRates 18-decimal FP values for rate tokens (e.g., yield-bearing), or FP(1) for standard tokens
+ * @param staticSwapFeePercentage 18-decimal FP value of the static swap fee percentage
+ * @param totalSupply The current total supply of the pool tokens (BPT)
+ * @param bptRate The current rate of a pool token (BPT) = invariant / totalSupply
+ * @param isPoolInitialized If false, the pool has not been seeded with initial liquidity, so operations will revert
+ * @param isPoolPaused If true, the pool is paused, and all non-recovery-mode state-changing operations will revert
+ * @param isPoolInRecoveryMode If true, Recovery Mode withdrawals are enabled, and live balances may be inaccurate
+ */
+struct GyroECLPPoolDynamicData {
```

Comment:
This is the same as the 2-CLP; technically this structure and corresponding function could be shared (i.e., both pool interfaces inherit an additional IGyroPoolDynamicData interface). Not sure if it's worth it.

Reply:
I don't think it's worth it; it just adds one more interface to the sandwich.
This is mostly for the SDK to consume, and we're just following the pattern of the existing pools.